Tasks/DeployVisualStudioTestAgent/TestAgentHelper.ps1

```diff
@@ -0,0 +1,205 @@
+function Get-TestAgentInstalledVersion($ProductVersion = "14.0")
+{
+	if($ProductVersion -eq "14.0") {
+		$InstalledCheckRegKey = ("Microsoft\DevDiv\vstf\Servicing\{0}\testagentcore" -f $ProductVersion)
+
+		$isProductExists = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName "Install"
+		$versionInstalled = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName "Version"
+
+		if(($isProductExists -eq "1") -and $versionInstalled) {
+			return $versionInstalled
+		}
+	}
+
+	if($ProductVersion -eq "15.0") {
+		$instance = Get-VisualStudio_15_0
+		if($instance) {
+			return $instance.VersionString;
+		}
+	}
+
+	return "0.0"
+}
+
+function Get-ProductEntry {
+	param
+	(
+		[string] $InstalledCheckRegKey,
+		[string] $InstalledCheckRegValueName
+	)
+
+	$installValue = ""
+	if ($InstalledCheckRegKey -and $InstalledCheckRegValueName)
+	{
+		$regPath = "REGISTRY::HKLM\Software\Wow6432Node\{0}" -f $InstalledCheckRegKey
+        ## Too bad, Get-ItemPropertyValue is not supported in PS 4.0 versions [Win8 and older world :(]
+        $installValue = (Get-ItemProperty -Path $regPath -ErrorAction SilentlyContinue).$InstalledCheckRegValueName
+		if(-not $installValue)
+		{
+			$regPath = "REGISTRY::HKLM\Software\{0}" -f $InstalledCheckRegKey
+            $installValue = (Get-ItemProperty -Path $regPath -ErrorAction SilentlyContinue).$InstalledCheckRegValueName
+		}
+	}
+
+	return $installValue
+}
+
+function Test-Container {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$LiteralPath)
+
+    Write-Verbose "Testing container: '$LiteralPath'"
+    if ((Test-Path -LiteralPath $LiteralPath -PathType Container)) {
+        Write-Verbose 'Exists.'
+        return $true
+    }
+
+    Write-Verbose 'Does not exist.'
+    return $false
+}
+
+function Get-VisualStudio_15_0 {
```

Comment:
Yes. but this will be execute throw remote powershell. can't be used like common task